app/helpers/audits_helper.rb

```diff
@@ -238,11 +238,26 @@ def main_object?(audit)
     main_objects_names.include?(type)
   end
 
-  def key_to_class(key, audit)
+  def find_auditable_type_class(audit)
     auditable_type = (audit.auditable_type == 'Host::Base') ? 'Host::Managed' : audit.auditable_type
+    auditable_type.constantize
+  end
+
+  def key_to_class(key, change, audit)
+    auditable_class = find_auditable_type_class(audit)
     association_name = key.gsub(/_id(s?)$/, '')
     association_name = association_name.pluralize if key =~ /_ids$/
-    auditable_type.constantize.reflect_on_association(association_name)&.klass
+    reflection_obj = auditable_class.reflect_on_association(association_name)
+    if reflection_obj.nil? && auditable_class.respond_to?('audit_hook_to_find_records')
+      auditable_class.send('audit_hook_to_find_records', key, change, audit)
```

Comment:
why calling it using send? if `auditable_class.respond_to?` returned true, it's safe to call `auditable_class.audit_hook_to_find_records()`

Reply:
Done! Modified code to fix failed tests.